tensorflow/python/ops/gradients_impl.py

```diff
@@ -1010,26 +1008,23 @@ def hessians(ys, xs, name="hessians", colocate_gradients_with_ops=False,
   hessians = []
   _gradients = gradients(ys, xs, **kwargs)
   for i, _gradient, x in zip(range(len(xs)), _gradients, xs):
-    # Ensure that x is a vector.
-    check_rank = check_ops.assert_rank(
-      x, 1, message='Cannot compute Hessian because element %d of `xs` does '
-      'not have rank one.' % i
+    # tricky change shape to one-dimension without graph branching
+    _gradient = array_ops.reshape(_gradient, [-1])
```

Comment:
Can we make sure that the original shape is reinstated before returning values? If I understand correctly, calculating the Hessian with respect to a tensor of shape `(a, b)` results in a Hessian of shape `(a * b, a, b)` using the current implementation?

In other words, the [reshape](https://github.com/tensorflow/tensorflow/pull/15308/files#diff-209b99b312c5b4e83f42002621103349R637) in the test should happen in the loop.

Reply:
Yes. current implementation return Hessian of shape (a * b, a, b) with respect to a tensor of shape(a, b).

But, I don't know which one is better.
I think simply should be keep the original tensor shape. 
So. I think shape (a, b, a, b) is best chosen. how about you?

Reply:
Yes, returning a shape `(a, b, a, b)` sounds perfect!